runtime/compiler/control/CompilationThread.cpp

```diff
@@ -5036,7 +5036,10 @@ void TR::CompilationInfo::recycleCompilationEntry(TR_MethodToBeCompiled *entry)
                    "recycling an improper entry\n");
    entry->_freeTag |= ENTRY_IN_POOL_NOT_FREE;
    if (entry->_numThreadsWaiting == 0)
+      {
```

Comment:
I would move the cleanup outside the `if` statement because otherwise you have another cleanup in another part of the code when entry moves `FREE` state (which you did)